include/ginkgo/core/solver/triangular.hpp

```diff
@@ -55,16 +55,68 @@ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 namespace gko {
 namespace solver {
 
-
-struct SolveStruct;
-
+enum class trisolve_type {
+    sparselib,
+    syncfree,
+    thinned,
+    block,
+    winv,
+    wvar,
+    level
+};
 
 /**
  * A helper for algorithm selection in the triangular solvers.
- * It currently only matters for the Cuda executor as there,
- * we have a choice between the Ginkgo syncfree and cuSPARSE implementations.
+ * It currently only matters for the Cuda executor,
+ * as we only there have different options.
  */
-enum class trisolve_algorithm { sparselib, syncfree };
+struct trisolve_strategy {
+    trisolve_type type;
+    uint8 thinned_m;
+    std::shared_ptr<std::vector<std::shared_ptr<trisolve_strategy>>>
+        block_inner;
```

Comment:
Let's assume we want to treat the block solves separately, so we still need this strategy type then? We should add some documentation for what `thinned_m` stands for